perf/perf_pmu.py

```diff
@@ -0,0 +1,267 @@
+            self.fail('Required sysfs files "sockets, coresperchip,\
+                      chipspersocket" not found.')
+
+        self._verify_lscpu_sysfs('%ssockets' % sysfs_file, self.psockets)
+        self._verify_lscpu_sysfs('%schipspersocket' % sysfs_file, self.pchips)
+        self._verify_lscpu_sysfs('%scoresperchip' % sysfs_file, self.pcorechips)
+
+        self._create_temp_user()
+        user_cmd = "su - test_pmu -c"
+        self._verify_lscpu_sysfs_test("%s 'cat %ssockets'"
+                                      % (user_cmd, sysfs_file), self.psockets)
+        self._verify_lscpu_sysfs_test("%s 'cat %schipspersocket'"
+                                      % (user_cmd, sysfs_file), self.pchips)
+        self._verify_lscpu_sysfs_test("%s 'cat %scoresperchip'"
+                                      % (user_cmd, sysfs_file), self.pcorechips)
+        self._remove_temp_user()
+
+        if self.pchips == "1" and self.psockets == "1" and self.pcorechips == "1":
+            output = process.system_output("dmesg").decode()
+            for line in output.split('\n'):
+                if 'rtas error: Error calling get-system-parameter' in line:
+                    self.fail("RTAS error occured")
+
+    def _check_count(self, event_type):
+        base_dir = "/sys/bus/event_source/devices/"
+        if not os.path.isdir(base_dir):
+            self.fail("sysfs events folder not found")
+
+        if not os.path.isdir(base_dir + event_type):
+            self.fail("sysfs %s folder not found" % event_type)
+
+        sys_fs_events = os.listdir(base_dir + event_type + '/events/')
+        if len(sys_fs_events) < 21:
+            self.fail("%s events folder contains less than 21 entries"
+                      % event_type)
+        self.log.info("%s events count = %s" % (event_type, len(sys_fs_events)))
+
+    def test_cpu_event_count(self):
+        # This test checks for the sysfs event_source directory and checks for
+        # cpu events count
+        self._check_count('cpu')
+
+    @skipIf(IS_POWER_NV or IS_KVM_GUEST, "This test is for PowerVM")
+    def test_hv_24x7_event_count(self):
+        # This test checks for the sysfs event_source directory and checks for
+        # hv_24x7  events count
+        self._check_count('hv_24x7')
+
+    @skipUnless(IS_POWER_NV, "This test is for PowerNV")
+    def test_imc_event_count(self):
+        self._check_count('core_imc')
+
+    @skipUnless(IS_POWER_NV, "This test is for PowerNV")
+    def test_thread_event_count(self):
+        self._check_count('thread_imc')
+
+    # TBD: collect only cpu, hv_24x7, imc
+    def _sysfs_pmu_entries(self, pmu_type):
+        sysfs_pmu = os.listdir('/sys/devices/')
+        sysfs_pmu_list = []
+        for entry in sysfs_pmu:
+            if pmu_type in entry:
+                sysfs_pmu_list.append(entry)
+        if not len(sysfs_pmu_list) > 0:
+            self.fail("%s PMUs not found in sysfs" % pmu_type)
+        else:
```

Comment:
```
    if not len(sysfs_pmu_list):
        self.fail()
    return ...
```

Reply:
else is not required here since the code is already past the condition and hence must always return a valid value